softkinetic_camera/src/softkinetic_start.cpp

```diff
@@ -1016,6 +1020,32 @@ int main(int argc, char* argv[])
     {
       device_index = atoi(argv[1]);
     }
+    else
+    {
+      nh.param<bool>("use_serial", use_serial, false);
+      if (use_serial)
+      {
+        nh.getParam("serial", serial);
```

Comment:
Yes, but only that you do not find the specified serial number.

Please check for empty `serial` string and add which serial you checked to the Error output.